src/app/models/prior_knowledge_reference.model.ts

```diff
@@ -1,6 +1,14 @@
-export interface PriorKnowledgeReferenceModel {
-  code: number;
+import {Module} from './module.model';
+
+export class PriorKnowledgeReference {
```

Comment:
I am wondering why you need to do a class with a single constructor. Use a fucking interface with proper initialization.